### PR TITLE
Sitecore Horizon images for SC 9.3

### LINKED ISSUE #243: Enhancement - Horizon Enabled Images and Docker-Compose Files for v.9.3.0.

### PR DESCRIPTION
**Pull request by afaniuolo**
This pull request implements Sitecore 9.3.0 XP images with Sitecore Horizon enabled.

Credits and references in documentation:
- Michael West's solution to run Docker containers with HTTPS (https://github.com/michaellwest/docker-https)
- Rob Ahnemann's monitor solution to automatically maintain DNS records for containers in hosts file (http://rockpapersitecore.com/2020/01/maintain-your-hosts-file-with-a-docker-container/).

This pull request resolves #243

### CODE REVIEW

**Comment by afaniuolo** [comment]
Good morning @bplasmeijer  and reviewers team!

Not sure if you had a chance to review this PR yet, but please let me know if any change is needed.

Thank you!
Alessandro

**Comment by mitikov** [comment]
@bplasmeijer @afaniuolo Folks, this one is pending for 2 months; could you please review that?

Thanks.

**Comment by sshushliapin** [comment]
Hi @afaniuolo, thanks for contributing the Horizon module!

I can successfully run `Build.ps1` from `master` (specifying SDN credentials). But when I run `Build.ps1` on this branch, I receive error regarding missing Publishing Module wdp package:

```
16:29:54:483: Required package found: 'C:\Users\serhii.shushliapin\Source\docker-images\packages\Sitecore Publishing Service 4.2.0-win-x64.zip'ForEach-Object : The property 'Sitecore Publishing Module 9.3.0.0 rev. r00546.2197.scwdp.zip' cannot be found on this object. Verify that the
property exists.
At C:\Users\serhii.shushliapin\Source\docker-images\modules\SitecoreImageBuilder\1.0.0\Public\Invoke-PackageRestore.ps1:69 char:17
+     $expected | ForEach-Object {
+                 ~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [ForEach-Object], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict,Microsoft.PowerShell.Commands.ForEachObjectCommand
```

AFAIKT, PS is an experimental module which should not be built by default (as I mentioned, `master` works fine and I did not enable experimental modules). I rebased to the latest master but nothing changed.
Have you experienced this before?

**Comment by afaniuolo** [comment]
Hi @sshushliapin, thank you for reporting it. I had accidentally removed the publishing package definition during my last rebase a couple weeks ago. It is fixed now.

**Comment by sshushliapin** [comment]
Yep, I see ;)

```
14:08:53:894: Downloading 'https://dev.sitecore.net/~/media/930F68120C454C868983EA04122E1FD4.ashx' to 'C:\Users\serhii.shushliapin\Source\docker-images\packages\Sitecore Horizon 9.3.0 for Azure.zip'...
Download complete
```

Building... I'll keep you informed. Thanks @afaniuolo!

**Comment by bic742** [comment]
Hi @afaniuolo,

I have the Horizon image built and configured via a compose, but Horizon does not appear to work. The launch pad link just goes back to the CM launch pad, and I am getting a 403 if I try to go to the host directly.

**Comment by bic742** [comment]
Chrome introduced the new restrictions on SameSite cookies. Having the three servers (cm, identity, authoring) using different domains means that the authentication process is not working as expected. Per the documentation for the install process, it is recommended that they all share a common domain. 

Can you update the hostnames to be `*.dev.local` (e.g. `cm.dev.local`, `authoring.dev.local`, `identity.dev.local`). You can use a wildcard to generate the cert, too.

This currently works in FF and Edge, but not on Chrome.

**Comment by afaniuolo** [comment]
@bic742, thank you for reporting this issue in Chrome. This should be fixed now. I have implemented your recommendation to use the same base domain (.dev.local.) for the application hostnames.

File: sitecore-packages.json [hunk 1]
```diff
@@ -1,231 +1,235 @@
 {
```
Comment (jeanfrancoislarente) [inline sitecore-packages.json:1]: Seems like you accidentally changed the spacing in this file?

Comment (afaniuolo) [reply sitecore-packages.json:1]: The spacing changed when I ran the `.\contributing\Sort-SitecorePackagesJson.ps1` script. I can manually fix it now.

Comment (afaniuolo) [reply sitecore-packages.json]: The spacing issue has been fixed now. Thank you for reporting it.

**Comment by bic742** [comment]
I am pulling latest from your fork, and will rebuild the image.

**Comment by bic742** [comment]
Confirmed it's all working in Chrome now.

@bplasmeijer @jeanfrancoislarente @joostmeijles @pbering - any chance this can be merged?

**Comment by sshushliapin** [comment]
+1. Works on my machine ;)

image

**Comment by afaniuolo** [comment]
Hi @bic742 ,

I merged the latest master branch in this PR. I re-tested it on my end and it looks good. Thank you for offering to test this PR again!

CC: @jeanfrancoislarente

**Review by jeanfrancoislarente — APPROVED** [review]

**Comment by jeanfrancoislarente** [comment]
Ran through the build and docker-compose test w/ instructions provided!

image

**Pull request merged**